Use Object property to replace the variable declaration using let

Heejae0704 opened the pull request:
Let을 대체하기 위해 Object의 프로퍼티를 이용했습니다.

Heejae0704 committed:
Use Object property to replace the variable decleration using let

Let을 대체하기 위해 Object의 프로퍼티를 이용했습니다.

heyjyu commented on src/index.js:
```diff
@@ -1,2 +1,60 @@
+          onClick={() => {
+            state.count += 1;
+            render();
+          }}
```
onClick에 들어가는 함수를 따로 역할에 맞는 함수명을 지어서 선언해주시면 더 좋을 것 같아요! (함수명으로 해당 함수의 역할을 알기가 더 쉬워집니다!)

Heejae0704 replied:
@heyjyu 역할에 맞는 함수명을 지어서 선언하였습니다. render 함수 내에서 선언하여 eslint의 no-use-before-define 룰을 설정하지 않아도 문제가 되지 않네요! ^^

heyjyu commented on src/index.js:
```diff
@@ -1,2 +1,60 @@
+            onClick={() => {
+              state.count = i;
+              render();
+            }}
```
https://github.com/CodeSoom/react-week1-assignment-1/pull/185/files#r916382605

Heejae0704 replied:
@heyjyu 버튼 만드는 로직도 별도 함수로 뺀 다음, 앞서 정의한 함수를 재활용했습니다.

heyjyu commented:
코드를 깔끔하게 잘 작성해주셨네요! 
let을 사용하지 말라는 의도는 const를 이용해서 render함수의 매개변수로 바뀌는 값을 받아서 값을 바꿀 때마다 새로 렌더하게 하는 것에 조금 더 가깝긴 해서 이렇게도 한 번 코드를 작성해보시면 도움이 될 것 같아요! 내일 해설강의가 올라오니 강의도 참고해주시면 좋을 것 같습니다!

Heejae0704 committed:
Improve overall code readability

1. count를 render 함수의 parameter로 변경
   - render할 때마다 변경되는 값이므로 같이 다니는 것이 합리적
   - 다른 코드에 영향을 미칠 수 있는 전역 변수 최소화
2. 코드를 기명함수로 정리
   - 재사용 가능
   - 코드가 하는 일을 명확하게 표현

Heejae0704 commented:
@heyjyu 네 알려주신 방법으로 하니 훨씬 깔끔하네요. count는 render할 때에 바뀌는 변수이니까 논리적으로도 이게 더 적합한 거 같고요, 다른 코드에 영향을 받을 수 있는 전역변수를 줄이는 방법이기도 해서 이 방식이 좋다고 생각합니다. 코드에 comment 주신 내용에 대해서도 수정했습니다. 감사합니다!
